### PR TITLE
Add documentation clarifying C# compiler behavior for static types

### PR DESCRIPTION
**Pull request by gafter**
related to #324

### CODE REVIEW

**Comment by gafter** [comment]
@AnthonyDGreen @theoy Can you please review this doc-only change? I think I just need one reviewer for docs.

File: docs/compilers/CSharp/Static Type Constraints.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Now as a separate matter the native compiler did not give an error when a static type was used as a type argument in a dynamic call, while Roslyn would give an error:
+
+```cs
+((dynamic)3).M<StaticType>()
+```
+
+This is the third bug (#511), and while Roslyn relies on the spec for its behavior, it is a breaking change from the native compiler. But as it turns out, this compat issue is fixed by the fix described above, because the type argument in a dynamic call is not checked against any type parameter constraints. So Roslyn now fails to diagnose this situation (i.e. doesn’t implement the required language rules) but at least it is compatible with the native compiler here.
+
+I’m on a horse.
```
Comment (mattwar) [inline docs/compilers/CSharp/Static Type Constraints.md]: What kind of horse?

Comment (gafter) [inline docs/compilers/CSharp/Static Type Constraints.md]: I wondered if anyone would really read this boring block of text.

Comment (khellang) [inline docs/compilers/CSharp/Static Type Constraints.md]: LOL :grin:

Comment (ilmax) [inline docs/compilers/CSharp/Static Type Constraints.md]: Hey @gafter, I'm reading this now :)

Comment (RichiCoder1) [inline docs/compilers/CSharp/Static Type Constraints.md]: :+1:

**Comment by gafter** [comment]
@RichiCoder1 Thank you!

File: docs/compilers/CSharp/Static Type Constraints.md (hunk 1 above)
Comment (parrt) [inline docs/compilers/CSharp/Static Type Constraints.md]: "Never look a gifted horse in the mouth"

**Pull request merged**